src/su.c

```diff
@@ -692,7 +693,7 @@ static /*@only@*/struct passwd * check_perms (void)
 		endpwent ();		/* close the old password databases */
 		endspent ();
 		pw_free (pw);
-		return check_perms ();	/* authenticate in the subsystem */
+		goto top; /* authenticate in the subsystem */
```

Comment:
Heh - I don't share the aversion to goto, however, since the only state that actually needs to be maintained between the calls is the tasks's chroot and curdir, an alternative implementation would be something like

```
static struct passwd *check_perms(void)
{
    struct passwd *pw = NULL;

    while (pw == NULL)
        pw = do_check_perms();
    return pw;
}
```
with do_check_perms() being the current code except return NULL instead of return check_perms() at the bottom.

Reply:
I have integrated the do_check_perms approach. This should also fix the Debian build.